class.jetpack-gutenberg.php

```diff
@@ -321,15 +369,34 @@ public static function get_availability() {
 			}
 		}
 
-		$unwhitelisted = array_fill_keys(
+		$unwhitelisted_blocks = array();
+		$all_registered_blocks = WP_Block_Type_Registry::get_instance()->get_all_registered();
+		foreach ( $all_registered_blocks as $block_name => $block_type ) {
+			if ( ! wp_startswith( $block_name, 'jetpack/' ) || isset( $block_type->parent ) ) {
+				continue;
+			}
+
+			$unprefixed_block_name = self::remove_extension_prefix( $block_name );
+
+			if( in_array( $unprefixed_block_name, self::$extensions ) ) {
```

Comment:
how can a block be registered without being whitelisted? shouldn't we prevent to register (or unregister) it if the extension is not whitelisted?